internal/transport/grpchttp2/http2bridge_test.go

```diff
@@ -0,0 +1,600 @@
+	if wf.Inc != 100 {
+		t.Errorf("ReadFrame(): Inc: got %d, want %d", wf.Inc, 1)
+	}
+}
+
+// Tests and verifies that the framer correctly merges Headers and Continuation
+// Frames into a single MetaHeaders Frame.
+func (s) TestBridge_ReadFrame_MetaHeaders(t *testing.T) {
+	fields := []hpack.HeaderField{
+		{Name: "foo", Value: "bar"},
+		{Name: "baz", Value: "qux"},
+	}
+
+	c := &testConn{}
+	enc := hpack.NewEncoder(c)
+	for _, field := range fields {
+		enc.WriteField(field)
+	}
+	half1 := c.wbuf[0 : len(c.wbuf)/2]
+	half2 := c.wbuf[len(c.wbuf)/2:]
+
+	c.rbuf = append(c.rbuf, 0, 0, byte(len(half1)), byte(FrameTypeHeaders), 0)
+	c.rbuf = appendUint32(c.rbuf, 1)
+	// Copy half of the data written by the encoder into the reading buf
+	c.rbuf = append(c.rbuf, half1...)
+
+	// Add Continuation Frame to test merging
+	c.rbuf = append(c.rbuf, 0, 0, byte(len(half2)), byte(FrameTypeContinuation), byte(FlagContinuationEndHeaders))
+	c.rbuf = appendUint32(c.rbuf, 1)
+	// Copy data written by the encoder into the reading buf
+	c.rbuf = append(c.rbuf, half2...)
+
+	f := NewFramerBridge(c, c, 0, nil)
+	fr, err := f.ReadFrame()
+	if err != nil {
+		t.Fatalf("ReadFrame(): %v", err)
+	}
+
+	mf, ok := fr.(*MetaHeadersFrame)
+	if !ok {
+		t.Errorf("ReadFrame(): Type: expected MetaHeadersFrame, got %T", fr)
+	}
+	if len(mf.Fields) != 2 {
+		t.Errorf("ReadFrame(): Fields: got %d, want %d", len(mf.Fields), 1)
+	}
+	for i, field := range fields {
+		if field.Name != mf.Fields[i].Name {
+			t.Errorf("ReadFrame(): Fields[%d].Name: got %q, want %q", i, mf.Fields[i].Name, field.Name)
+		}
+		if field.Value != mf.Fields[i].Value {
+			t.Errorf("ReadFrame(): Fields[%d].Value: got %q, want %q", i, mf.Fields[i].Value, field.Value)
+		}
+	}
+}
+
+// Tests and verifies that a Data Frame is correctly written.
+func (s) TestBridge_WriteData(t *testing.T) {
+	c := &testConn{}
+	wantData := "test data"
+	testBuf := [][]byte{[]byte("test"), []byte(" data")}
+	f := NewFramerBridge(c, c, 0, nil)
+	f.WriteData(1, false, testBuf...)
+	wantHdr := wantHeader{
+		wantSize:      len(wantData),
+		wantFrameType: FrameTypeData,
+		wantFlag:      0,
+		wantStreamID:  1,
+	}
+	if errs := checkWrittenHeader(c.wbuf[0:9], wantHdr); len(errs) > 0 {
+		t.Errorf("WriteData():\n%s", errors.Join(errs...))
+	}
+	if string(c.wbuf[9:]) != wantData {
+		t.Errorf("WriteData(): Data: got %q, want %q", string(c.wbuf[9:]), wantData)
+	}
+}
+
+// Tests and verifies that a Headers Frame and all its flag permutations are
+// correctly written.
+func (s) TestBridge_WriteHeaders(t *testing.T) {
+	tests := []struct {
+		endStream  bool
+		endHeaders bool
+	}{
+		{endStream: false, endHeaders: false},
+		{endStream: true, endHeaders: false},
+		{endStream: false, endHeaders: true},
+		{endStream: true, endHeaders: true},
+	}
+	wantData := "test data"
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("endStream=%v, endHeaders=%v", test.endStream, test.endHeaders), func(t *testing.T) {
```

Comment:
Oh sorry, should have been more clear. Also, define a field in the table, `desc` or `name` which contains a description or a name for the subtest, and here, you can simply do `t.Run(test.name, ...)`.

See here for more info: https://g3doc.corp.google.com/go/g3doc/style/decisions.md?cl=head#table-driven-tests

Reply:
Sorry for not catching that! Done.